src/UnoFlipModel.java

```diff
@@ -299,10 +300,15 @@ public void playTurn(int btnIndex){        // add a
             chosenCardIndex = btnIndex;
 
             if (chosenCardIndex == -1) { // SELF DRAW ONE
-                turnSeqs.get(14).executeSequence(null);
-                status = " ";
-                notifyViews();
-                turnFinished = true;
+                if (validSelfDrawOne()){
+                    turnSeqs.get(14).executeSequence(null);
+                    status = " ";
+                    notifyViews();
+                    turnFinished = true;
+                } else{
+                    status = "YOU HAVE PLAYABLE CARD";
+                    notifyViews();
```

Comment:
you notify views in both the if and else blocks. should be at the end of the conditional, one single line.